Cargo.toml

```diff
@@ -1,6 +1,9 @@
 [workspace]
 # Use the new resolver to prevent dev-deps and build-deps from enabling debugging or test features in production.
 members = [
+  "txpool",
+  "interfaces",
+  "block-producer",
```

Comment:
We should keep the block producer independent of this PR unless there's some traits we require from it.